test/test_group.cpp

```diff
@@ -2315,4 +2315,60 @@ TEST(Group_UniqueColumnKeys)
     CHECK_NOT_EQUAL(col_foo, col_bar);
 }
 
+ONLY(Group_ArrayCompression_Correctness)
+{
+    GROUP_TEST_PATH(path);
```

Comment:
Added encode/decode logic to `Array::insert` path (probably we should do this in the end in order to avoid exposing compress/decompress methods through Group, but inserting becomes more expensive.)